src/zope/interface/declarations.py

```diff
@@ -111,6 +111,53 @@ def __add__(self, other):
     __radd__ = __add__
 
 
+class _ImmutableDeclaration(Declaration):
+    # A Declaration that is immutable. Used as a singleton to
+    # return empty answers for things like ``implementedBy``.
+    # We have to define the actual singleton after normalizeargs
+    # is defined, and that in turn is defined after InterfaceClass and
+    # Implements.
+
+    __slots__ = ()
+
+    __instance = None
+
+    def __new__(cls):
+        if _ImmutableDeclaration.__instance is None:
+            _ImmutableDeclaration.__instance = object.__new__(cls)
+        return _ImmutableDeclaration.__instance
+
+    def __reduce__(self):
+        return "_empty"
+
+    @property
+    def __bases__(self):
+        return ()
+
+    @__bases__.setter
+    def __bases__(self, new_bases):
+        if new_bases:
```

Comment:
Why allowing setting at all? When is it the case to set an empty base here?

Reply:
The constructor of the base class sets the bases.

Reply:
Ok, got it! Probably a short comment in code helps to reduce confusion of our future-selfs and others reading the code :) ?

Reply:
Sure!